frontend/src/components/recentTransactions.jsx

```diff
@@ -57,12 +51,101 @@ const recentTransactions = () => {
     fetchTransactions();
   }, [currentUser]);
 
+  useEffect(() => {
+    const fetchConfidenceScores = async () => {
+      if (!transactions.length || !currentUser) return;
+
+      const token = await currentUser.getIdToken();
+      const scores = {};
+      const synonymInfo = {};
+
+      // gets unique merchants
+      const merchantNames = [...new Set(transactions.map((tx) => tx.merchant))];
+
+      // gets confidence score for each merchant
+      for (const merchantName of merchantNames) {
+        try {
+          const response = await fetch(
```

Comment:
usually an anti-pattern doing await in a sequential loop, this can be slow if there are many merchants, leading to poor user experience.

Instead, fire requests in parallel using `Promise.all` if the API supports it, or optimize the backend to allow batch requests.